src/components/pages/doc.js

```diff
@@ -133,40 +133,39 @@ class Doc extends React.Component {
     const { t, user } = this.props;
     const { revisions, currentRevision } = this.state;
 
-    let revisionPicker = null;
-
-    if (hasUserPermission(user, permissions.MANAGE_DOC_REVISIONS) && revisions.length > 1) {
-      const marks = revisions.reduce((accu, item, index) => {
-        accu[index] = index === 0 || index === revisions.length - 1 ? (index + 1).toString() : '';
-        return accu;
-      }, {});
-
-      const currentRevisionIndex = revisions.indexOf(currentRevision);
-      const isCurrentRevisionLatestRevision = currentRevisionIndex === revisions.length - 1;
-
-      revisionPicker = (
-        <div className="DocPage-revisionPicker">
-          <div className="DocPage-revisionPickerLabel">{t('revision')}:</div>
-          <div className="DocPage-revisionPickerSlider">
-            <Slider
-              min={0}
-              max={revisions.length - 1}
-              value={currentRevisionIndex}
-              step={null}
-              marks={marks}
-              onChange={this.handleIndexChanged}
-              tipFormatter={this.formatRevisionTooltip}
-              />
-          </div>
-          <div className="DocPage-revisionPickerButtons">
-            <Button
-              className="DocPage-revisionPickerButton"
-              type="primary"
-              icon={<PaperClipOutlined />}
-              onClick={this.handlePermalinkRequest}
-              >
-              {t('permalink')}
-            </Button>
+    const marks = revisions.reduce((accu, item, index) => {
+      accu[index] = index === 0 || index === revisions.length - 1 ? (index + 1).toString() : '';
+      return accu;
+    }, {});
+
+    const currentRevisionIndex = revisions.indexOf(currentRevision);
+    const isCurrentRevisionLatestRevision = currentRevisionIndex === revisions.length - 1;
+    const canRestoreRevisions = hasUserPermission(user, permissions.RESTORE_DOC_REVISIONS);
+
+    const revisionPicker = (
+      <div className="DocPage-revisionPicker">
+        <div className="DocPage-revisionPickerLabel">{t('revision')}:</div>
+        <div className="DocPage-revisionPickerSlider">
+          <Slider
+            min={0}
+            max={revisions.length - 1}
+            value={currentRevisionIndex}
+            step={null}
+            marks={marks}
+            onChange={this.handleIndexChanged}
+            tipFormatter={this.formatRevisionTooltip}
+            />
+        </div>
+        <div className="DocPage-revisionPickerButtons">
+          <Button
+            className="DocPage-revisionPickerButton"
+            type="primary"
+            icon={<PaperClipOutlined />}
+            onClick={this.handlePermalinkRequest}
+            >
+            {t('permalink')}
+          </Button>
+          {canRestoreRevisions && (
             <Button
```

Comment:
BTW, there is a `<Restricted />` component, it help with readability:

~~~
<Restricted to={permissions.RESTORE_DOC_REVISIONS}>
    <Button ... />
</Restricted>
~~~

Maybe you want to take it into consideration.

Reply:
I definitely want to use this, thanks!
Updated.